### ISSUE TITLE
No config file

### ISSUE THREAD

**Issue by ghost** [zoltantamasvajda/openttd-server#3]
I installed the snap on my Raspberry Pi 2, however I cannot find any config file. Inside /snap/openttd-server/current I only have these folders: 
-   lib 
-   meta
-   snap 
-   usr 
There is no .config folder even if I do ls -a and I tried searching for openttd.cfg across the system and it isn't there.

If i try to create the .config folder with mkdir it says it's a read only file system

**Comment by Tom5421** [comment]
I've tried on multiple platforms and I am getting a similar issue, the directory (/root/snap/openttd-server/current /.config/openttd) exists however it does not create the config file.

I've tried manually placing a pre-defined config file here and it does not work either.

I've tried starting and stopping server multiple times and the file does not get generated. 

I have tested on Digital Ocean VPS Droplet, VirtualBox Virtual Machine and Rasperry Pi 4. Appears the issue is not system dependant.

**Comment by ghost** [comment]
Yes the Snap file system is read only so it does not create the config file.

**Comment by Tom5421** [comment]
So how can I use the snap with a configuration file?

The description says "The configuration file can be found under /root/snap/openttd-server/current/.config/openttd/openttd.cfg" but the file is not created here.

**Comment by ghost** [comment]
I didn't figure it out but I was able to run the server just with default
settings. Probably using the snap version isn't the best way.

**Comment by Tom5421** [comment]
It seems like the snap is broken or not running correctly which is frustrating. 

I eventually got an OpenTTD server working with Docker, but snap can be a lot quicker and more convenient if I quickly want to spin up a server on a VPS for example.

**Comment by zoltantamasvajda** [comment]
Wow, I did not realise there are so many people using this 😅. It seems like at some point openttd got an update that broke things. I have tracked down and fixed the issue and published a new update. 

I have updated the readme file with new instructions, but I'll also paste them here for convenience:

tldr.: OpenTTD no longer autogenerates the cfg file on first start. It has to be manually created using the `saveconfig` rcon command.

## Configuring: 

### Using rcon

 The recommended way to configure the server is using [rcon](https://wiki.openttd.org/en/Manual/Dedicated%20server#controlling-the-server-with-rcon). The default rcon password is `default` and can be changed by running:

`sudo snap set openttd-server rcon-passwd=YourChosenPassword`

### Editing the config file

Alternatively the configuration file can be found under `/root/snap/openttd-server/current/.config/openttd/openttd.cfg` and can be used the [usual way](https://wiki.openttd.org/en/Archive/Manual/Settings/Openttd.cfg) (root filesystem access required).

In order for the changes to persist the server has to be stopped using `sudo snap stop openttd-server` before editing the config file. After the edits are saved the server can be restarted using `sudo snap start openttd-server`.

By default the file only contains the default rcon password. In order to generate a full settings file:
1. Start the server
2. Join the server
3. Open the console using the `~` key
4. run `rcon default saveconfig`

This will populate the config file with the default settings which can then be edited as needed.
